src/main/java/com/diamondfire/helpbot/bot/command/impl/other/util/SolvedCommand.java

```diff
@@ -76,8 +81,11 @@ public void run(CommandEvent event) {
         // Apply the solved tag, other behavior handled by PostAppliedTagsEvent.
         ForumTag solvedTag = threadChannel.getParentChannel().asForumChannel().getAvailableTagById(HelpBotInstance.getConfig().getHelpChannelSolvedTag());
         ArrayList<ForumTag> appliedTags = new ArrayList<>(threadChannel.getAppliedTags());
-        if (!appliedTags.contains(solvedTag)) appliedTags.add(solvedTag);
-        
+        if (appliedTags.contains(solvedTag)) {
+            appliedTags.remove(solvedTag);
+            threadChannel.getManager().setAppliedTags(appliedTags).queue();
+        }
+        appliedTags.add(solvedTag);
```

Comment:
What is the point of this If statement if you're just re-adding it anyway

Reply:
to trigger the apply tag event that handles renaming and sending the message

Reply:
i could move it all to a cetric class but i dont want this whole solved thing to leave a big footprint